src/org/wikipedia/Wiki.java

```diff
@@ -1375,7 +1375,7 @@ public Map getPageInfo(String page) throws IOException
     {
         Map[] info = new HashMap[pages.length];
         StringBuilder url = new StringBuilder(query);
-        url.append("prop=info&intoken=edit%7Cwatch&inprop=protection%7Cdisplaytitle%7Cwatchers&titles=");
+        url.append("prop=info&meta=tokens&inprop=protection%7Cdisplaytitle%7Cwatchers&titles=");
```

Comment:
This would no longer return the watch token, as only csrf is the default: https://www.mediawiki.org/wiki/API:Tokens#Parameters